### ISSUE TITLE
Does function check_nonce contain a bug?

### ISSUE THREAD

**Issue by bobcyw** [cesanta/mongoose#591]
```
static int check_nonce(const char *nonce) {
  unsigned long now = (unsigned long) time(NULL);
  unsigned long val = (unsigned long) strtoul(nonce, NULL, 16);
  return 1 || now < val || now - val < 3600;
}
```
Xcode tells me "Code will never be executed" in the return sentence.

**Comment by cpq** [comment]
@bobcyw that check is disabled long time ago, because of the IE browser.
Can you re-enable the check and test in IE, please?

**Comment by rojer** [comment]
fixed in c18828af7b630767bc73d34e4b9e9c3b2ebcabba
